src/V3EmitCImp.cpp

```diff
@@ -922,23 +922,39 @@ void V3EmitC::emitcImp() {
     // Make parent module pointers available.
     const EmitCParentModule emitCParentModule;
     std::list<std::deque<AstCFile*>> cfiles;
+    std::list<std::future<void>> futures;
 
     // Process each module in turn
     for (const AstNode* nodep = v3Global.rootp()->modulesp(); nodep; nodep = nodep->nextp()) {
         if (VN_IS(nodep, Class)) continue;  // Imped with ClassPackage
         const AstNodeModule* const modp = VN_AS(nodep, NodeModule);
         cfiles.emplace_back();
-        EmitCImp::main(modp, /* slow: */ true, cfiles.back());
+        auto& slow = cfiles.back();
+        futures.push_back(V3ThreadPool::s().enqueue<void>(
+            [modp, &slow]() { EmitCImp::main(modp, /* slow: */ true, slow); }));
         cfiles.emplace_back();
-        EmitCImp::main(modp, /* slow: */ false, cfiles.back());
+        auto& fast = cfiles.back();
+        futures.push_back(V3ThreadPool::s().enqueue<void>(
+            [modp, &fast]() { EmitCImp::main(modp, /* slow: */ false, fast); }));
     }
 
     // Emit trace routines (currently they can only exist in the top module)
     if (v3Global.opt.trace() && !v3Global.opt.lintOnly()) {
         cfiles.emplace_back();
-        EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ true, cfiles.back());
+        auto& slow = cfiles.back();
+        futures.push_back(V3ThreadPool::s().enqueue<void>([&slow]() {
+            EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ true, slow);
+        }));
         cfiles.emplace_back();
-        EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ false, cfiles.back());
+        auto& fast = cfiles.back();
+        futures.push_back(V3ThreadPool::s().enqueue<void>([&fast]() {
+            EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ false, fast);
+        }));
+    }
+    // Wait for futures
+    while (!futures.empty()) {
+        V3ThreadPool::s().waitForFuture(futures.front());
+        futures.pop_front();
```

Comment:
Make a function in ThreadPool to do this, e.g. V3ThreadPool::s()::waitForFutures(futures)?